src/main/java/org/radarcns/management/service/mapper/ClientDetailsMapperImpl.java

```diff
@@ -52,6 +55,8 @@ public ClientDetails clientDetailsDTOToClientDetails(ClientDetailsDTO detailsDTO
         if (detailsDTO.getAutoApprove()) {
             result.setAutoApproveScopes(Collections.unmodifiableSet(detailsDTO.getScope()));
         }
+        result.setAdditionalInformation(Collections.unmodifiableMap(detailsDTO
```

Comment:
It seems that detailsDTO.getAdditionalInformation() can return null. In that case Collections.unmodifiableMap will produce a NPE. Same for other fields properties.

Reply:
Good point, added the necessary null checks in this class.